src/modules/middleware/roleCheckGuard.js

```diff
@@ -0,0 +1,34 @@
+const Boom = require('boom');
+const { authService } = require('../authorization/authService');
+const { ShopRepository } = require('../shops/shop.repository');
+const { checkRolePermissions } = require('../../libs/checkRolePermissions');
+
+function forbidden(ctx) {
+  const error = Boom.forbidden('you are not allowed to perform this action');
+  const errorData = error.output.payload;
+  ctx.status = errorData.statusCode;
+  ctx.body = errorData;
+  return null;
+}
+
+const roleCheckGuard = async (ctx, next, action) => {
+  const token = ctx.request.header.authorization.split(' ')[1];
+  const shopUUID = ctx.params.id;
+  const user = await authService.getUserByToken(token);
+  const admins = await ShopRepository.getAdminsByUUID(shopUUID);
+  const isAdmin = admins.find((admin) => admin.id === user.id);
```

Comment:
checkRolePermissions should verify if user has access to the action, while this method shoul only collect data from DB and pass it to checkRolePermissions